tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/InstrumentationDefinitionsGenerator.cs

```diff
@@ -3,33 +3,43 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
+#pragma warning disable SA1204 // StaticElementsMustAppearBeforeInstanceElements
 using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading;
 using Datadog.Trace.ClrProfiler;
 using Datadog.Trace.SourceGenerators.Helpers;
 using Datadog.Trace.SourceGenerators.InstrumentationDefinitions;
 using Datadog.Trace.SourceGenerators.InstrumentationDefinitions.Diagnostics;
+using Helpers;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Text;
+using static AspectsDefinitionsGenerator;
 
 /// <inheritdoc />
 [Generator]
-public class InstrumentationDefinitionsGenerator : IIncrementalGenerator
+public class InstrumentationDefinitionsGenerator : IncrementalGeneratorBase
 {
     private const string InstrumentedMethodAttribute = "Datadog.Trace.ClrProfiler.InstrumentMethodAttribute";
     private const string AdoNetInstrumentAttribute = "Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.AdoNetClientInstrumentMethodsAttribute";
     private const string AdoNetTargetSignatureAttribute = AdoNetInstrumentAttribute + ".AdoNetTargetSignatureAttribute";
 
     /// <inheritdoc />
-    public void Initialize(IncrementalGeneratorInitializationContext context)
+    public override void Initialize(IncrementalGeneratorInitializationContext context)
     {
+        // Get path of the compiling project using the placeholder additional file
+        var placeholderFile = RegisterPlaceholder(context);
```

Comment:
Currently we're duplicating the work for both of these generators - I wonder if we should just combine them (in a subsequent PR) so that we reduce some of the work done by these common functions?

Reply:
Yeah, I think it's worth a shot later